libshaderc_spvc/src/spvcir_pass.h

```diff
@@ -25,16 +25,31 @@ namespace opt {
 // this WIP pass generates spvc IR and does not throw exceptions
 class SpvcIrPass : public Pass {
  public:
-  SpvcIrPass(spirv_cross::ParsedIR& ir) { ir_ = ir; }
-  const char* name() const override { return "spvc-if-pass"; }
+  SpvcIrPass(spirv_cross::ParsedIR &ir);
+  const char *name() const override { return "spvc-if-pass"; }
   Status Process() override;
 
   IRContext::Analysis GetPreservedAnalyses() override {
     return IRContext::kAnalysisNone;
   }
 
+  void genSpirvCrossIR(Instruction *inst);
+
  private:
-  spirv_cross::ParsedIR ir_;
+  spirv_cross::ParsedIR *ir_;
+  spirv_cross::SPIRFunction *current_function_ = nullptr;
+  spirv_cross::SPIRBlock *current_block_ = nullptr;
+
+  // copied from spirv-cross (class Parser)
```

Comment:
Say what this does:

// Adds an instruction to the SPIRV-Cross IR, for the given result Id and with given type and arguments.
// Returns the constructed instruction.